### ISSUE TITLE
eth core dump with bind: Cannot assign requested address

### ISSUE THREAD

**Issue by ethers** [ethereum/cpp-ethereum#1381]
Just built develop e01002b.  eth core dumps (AZ seems to launch fine).

```
./eth/eth -j -m on -v 9
Ethereum (++) 0.9.1
  Code by Gav Wood et al, (c) 2013, 2014, 2015.
  Based on a design by Vitalik Buterin.

*N* [ 15:40:54 | main ] IP Address: <address>  = [LOCAL]
*N* [ 15:40:54 | main ] Id: a38bc91fcded45ed499337411ab7ebf804f4e566da6a911b323ed21765cba723d37ac65b0db0989d0de27a3f2e67fdc89da2b02a57c7b2817a0923e115e8cc7f
*** [ 15:40:54 | main ] Opened blockchain DB. Latest: 8d4791d76b6f07497e2b4e6cb1ffa79b593e66814a3a99895e78402e800c27e2
*** [ 15:40:54 | main ] Opened state DB.
*** [ 15:41:17 | main ] noteChanged( { 0000000000000000000000000000000000000000000000000000000000000000 } )
--- [ 15:41:17 | main ] startWorking for thread eth
--- [ 15:41:17 | main ] Spawning eth
Transaction Signer: f52a423ecad0b6c70b5adf69edbe51b9db11c7c9
Mining Benefactor: f52a423ecad0b6c70b5adf69edbe51b9db11c7c9
--- [ 15:41:17 | main ] startWorking for thread p2p
--- [ 15:41:17 | main ] Spawning p2p
--- [ 15:41:17 | p2p ] startWorking for thread ethsync
--- [ 15:41:17 | p2p ] Spawning ethsync
--- [ 15:41:17 | p2p ] startWorking for thread anon
--- [ 15:41:17 | p2p ] Spawning anon
--- [ 15:41:17 | main ] startWorking for thread eth
--- [ 15:41:17 | main ] startWorking for thread miner-0
--- [ 15:41:17 | main ] Spawning miner-0
--- [ 15:41:17 | main ] startWorking for thread miner-1
--- [ 15:41:17 | main ] Spawning miner-1
--- [ 15:41:17 | main ] startWorking for thread miner-2
--- [ 15:41:17 | main ] Spawning miner-2
--- [ 15:41:17 | main ] startWorking for thread miner-3
--- [ 15:41:17 | main ] Spawning miner-3
*** [ 15:41:17 | eth ] noteChanged( { 0000000000000000000000000000000000000000000000000000000000000000 } )
*** [ 15:41:19 | p2p ] UPnP device: http://<address>:<port>/InternetGatewayDevice.xml [st: urn:schemas-upnp-org:device:InternetGatewayDevice:1 ]
*N* [ 15:41:19 | p2p ] Punched through NAT and mapped local port 30303 onto external port 30303 .
*N* [ 15:41:19 | p2p ] External addr: <address>
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::system::system_error> >'
  what():  bind: Cannot assign requested address
[1]    18565 abort (core dumped)  ./eth/eth -j -m on -v 9
```

**Comment by subtly** [comment]
This will occur in some cases if an instance of ethereum (or something else) is already running and listening on the default peer port or json-rpc port. (ethereum: 30303, json-rpc: 8080).

**Comment by ethers** [comment]
@subtly thanks for reminder; before filing I did check and didn't have any other ethereum stuff running (also, Alethzero would not have been able to launch either if some of the common ports were in use, but AZ is running fine so I am using it instead for now)

**Comment by ethers** [comment]
looks fixed on develop e904462

**Comment by aperseghin** [comment]
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::system::system_error> >'
  what():  bind: Cannot assign requested address
Aborted (core dumped)
-------------------------------
eth version 0.9.1
eth network protocol version: 58
Client database version: 11784
Build: linux/g++/int/Release
